File: AdventJS/2025/reto 8/8.js
/**
 * @param {string} toy - The toy to find the first unique one letter
 * @returns {string} The first unique letter in the toy
 */
function findUniqueToy(toy) {
  let letraQuitada = "";
  let nuevo = "";

  if (toy.length <= 0) {
    return "";
  } 

  for (let i = 0; i < toy.length; i++) {
    letraQuitada = toy.slice(i, i+1);
    nuevo = toy.slice(0, i) + toy.slice(i + 1);
    if (!nuevo.toLowerCase().includes(letraQuitada.toLowerCase())) {
      return letraQuitada;
    }
  }
  return ''
}

File: AdventJS/2025/reto 8/8.py
def find_unique_toy(toy: str) -> str:
  largo = len(toy)
  
  if largo <= 0: return ""
  
  for i in range(largo):
    letra_quitada = toy[i:i+1]
    nuevo = toy[:i] + toy[i+1:]
    if letra_quitada.lower() not in nuevo.lower():
      return letra_quitada

  return ""